raven/raven.go

```diff
@@ -268,3 +306,29 @@ func (T *transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	defer timer.Stop()
 	return T.httpTransport.RoundTrip(req)
 }
+
+type EventEncoder interface {
```

Comment:
The nice thing about doing it with httptest is that the same approach could be used for all the tests by designing a simple mock server component. Otherwise we need to obscure the code with abstractions that don't have much use beyond making some things easier to test.

It's a desirable quality to have if you want to have modular unit testable code, but I think since this pretty much a single purpose type it's better to lean more towards the black box testing approach which also verifies things at the final output instead of an intermediate stage.